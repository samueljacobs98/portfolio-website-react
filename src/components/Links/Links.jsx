import "./Links.scss";
import linkLines from "../../assets/svgs/link-lines.svg";
import insta from "../../assets/svgs/instagram-icon.svg";
import github from "../../assets/svgs/github-icon.svg";
import linkedIn from "../../assets/svgs/linkedin-icon.svg";

const Links = () => {
  return (
    <div className="links">
      <a
        href="https://www.linkedin.com/in/samuel-j-66a84ba2/"
        target="_blank"
        rel="noreferrer"
      >
        <img className="links__icon" src={linkedIn} alt="link to linked in" />
      </a>
      <a
        href="https://github.com/samueljacobs98"
        target="_blank"
        rel="noreferrer"
      >
        <img className="links__icon" src={github} alt="link to github" />
      </a>
      <a
        href="https://www.instagram.com/_samuel.jacobs_"
        target="_blank"
        rel="noreferrer"
      >
        <img className="links__icon" src={insta} alt="link to instagram" />
      </a>
      <img
        className="links__lines"
        src={linkLines}
        alt="lines leading to social links"
      />
    </div>
  );
};

export default Links;
