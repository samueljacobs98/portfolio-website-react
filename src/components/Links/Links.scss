.links {
  position: fixed;
  bottom: 0;
  right: 0;
  margin-right: 15px;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: center;
  &__lines {
    height: 40vh;
  }
  &__icon {
    margin: 15px;
  }
  &__icon:hover {
    cursor: pointer;
  }
}

@media screen and (max-width: 1260px) {
  .links {
    flex-direction: row;
    &__lines {
      display: none;
    }
  }
}

@media screen and (max-width: 850px) {
  .links {
    display: none;
  }
}
@media screen and (max-width: 850px) {
  .contact {
    .links {
      width: 100%;
      display: flex;
      position: absolute;
      justify-content: center;
      margin-right: 0;
    }
  }
}
