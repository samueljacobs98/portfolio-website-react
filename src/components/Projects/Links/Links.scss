@use "../../../assets/scss/variables" as *;

.links {
  display: flex;
  justify-content: space-between;
  gap: 30px;
  border: 2px solid $color-white;
  padding: 10px 15px;
  &__container {
    display: flex;
    align-items: center;
    justify-content: center;
    .links__link:hover {
      cursor: pointer;
    }
  }
}
