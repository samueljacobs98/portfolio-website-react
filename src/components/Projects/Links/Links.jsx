import github from "../../../assets/svgs/github.svg";
import preview from "../../../assets/svgs/preview.svg";
import "./Links.scss";

const Links = () => {
  return (
    <div className="links">
      <div className="links__container">
        <p className="links__text">Code:</p>
        <img className="links__link" src={github} alt="" />
      </div>
      <div className="links__container">
        <p className="links__text">Code:</p>
        <img className="links__link" src={preview} alt="" />
      </div>
    </div>
  );
};

export default Links;
