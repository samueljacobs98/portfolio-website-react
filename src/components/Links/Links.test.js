import { render } from "@testing-library/react";
import Links from "./Links";

test("should match snapshot", () => {
  const { container } = render(<Links />);
  expect(container).toMatchSnapshot();
});
