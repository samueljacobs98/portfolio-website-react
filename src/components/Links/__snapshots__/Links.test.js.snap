// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="links"
  >
    <a
      href="https://www.linkedin.com/in/samuel-j-66a84ba2/"
    >
      <img
        alt="link to linked in"
        class="links__icon"
        src="linkedin-icon.svg"
      />
    </a>
    <a
      href="https://github.com/samueljacobs98"
    >
      <img
        alt="link to github"
        class="links__icon"
        src="github-icon.svg"
      />
    </a>
    <a
      href="https://www.instagram.com/_samuel.jacobs_"
    >
      <img
        alt="link to instagram"
        class="links__icon"
        src="instagram-icon.svg"
      />
    </a>
    <img
      alt="lines leading to social links"
      class="links__lines"
      src="link-lines.svg"
    />
  </div>
</div>
`;
